packages/babel-parser/src/options.ts

```diff
@@ -41,6 +42,9 @@ export const defaultOptions: Options = {
   // When enabled, a return at the top level is not considered an
   // error.
   allowReturnOutsideFunction: false,
+  // When enabled, new.target at the top level is not considered an
+  // error.
```

Comment:
The current implementation also allows `new.target` within an arrow function. If it is desired, can you update the note and add a new test?

Reply:
To clarify: Are you saying that "top level" here is the wrong words to use because inside an arrow function is not top level?

Reply:
Given that
```js
function fn() {
  () => new.target
} 
```
is valid, I would expect `() => new.target` to be valid with this option, right?

Reply:
I don't think "top level" is wrong. But maybe it is incomplete in this case, since `() => new.target` will also be allowed.

Reply:
Yes, intent with this option was to allow `new.target` anywhere. If the file was wrapped in a function e.g. if it's a CommonJS file, I'm not aware of anywhere where `new.target` can't be used - because there's always a function above it.

Have added tests for `new.target` in an arrow function and updated the description of the option accordingly.